### PR TITLE
Cleaning House and Handing Over in 2022

### PR DESCRIPTION
**Pull request by dikunong**
**If no one reviews, I'll merge this PR on Wednesday evening.**

Give a frontend engineer a free afternoon at work and oh boy!

- Update to Gatsby v4 (from v2, oops!) and its dependencies
- Update files we use from https://github.com/gatsbyjs/gatsby-starter-default
- Delete unused pages and assets
- Refactor lots of repeated code into components
  - Most importantly `Banner`, now identical between **Index** and **Sponsor Us**, yay
- Begone, class components (one exception, see below)
- Some styling consistency improvements
- Add 2021 to Previous Shows and remove 2021 marketing material
- Add sexy 404 page
- Add maintenance instructions to the README for @f-fromont 

_NOTE: I didn't touch the **Previous Shows** page because that page is in need of a larger overhaul. Maybe I'll get around to it, maybe I won't._

This codebase is definitely still a mess (it's definitely not on my CV anymore haha) but hopefully it's now in a slightly better place :)

### CODE REVIEW

File: src/pages/404.js [hunk 1]
```diff
@@ -1,13 +1,26 @@
...
+        color: "white",
+        display: "grid",
+        flex: `1`,
+        placeContent: "start center",
+        marginTop: "120px",
+        textAlign: "center",
+      }}
+    >
+      <h1 style={{ fontSize: `48px` }}>NOT FOUND</h1>
+      <p>Like the clothing in Boys' Dance, this page doesn&#39;t exist.</p>
```
Comment (f-fromont) [inline src/pages/404.js:22]: lmao

**Review by f-fromont — APPROVED** [review]
I had a look through all the files, and the changes LGTM

**Comment by alex-ruddell** [comment]
Also looks good to me, but that being said I haven't touched React in ages :P 

Re: the Previous Shows page, I'd love to find the time to rework this at some point haha, maybe when I've got a spare afternoon at work..

Looks great man, aside from the naked boys, which looks AMAZING. Lgtm.

File: src/pages/404.js (hunk 1 above)
Comment (dikunong) [reply src/pages/404.js:22]: 🕺

**Pull request merged**